implement logic

kyeah01 opened the pull request:
@kimevanjunseok
하하 잘못열었꾼

kyeah01 committed:
implement logic

kyeah01 committed:
replace javafxapplication to class

kyeah01 committed:
add main method

kimevanjunseok commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,46 @@
+
+public class Main {
+    public static void main (String[] args) {
+        Scanner sc = new Scanner(System.in);
```
만약 자주쓰인다면 맴버 변수로 빼는 게 좋아용
```
public class Main {
    private final static Scanner sc = new Scanner(System.in);

    public static void main (String[] args) {
    }
}
이런식으로
```

kimevanjunseok replied:
final과 static은 공부하면 좋을거얌

kyeah01 replied:
final
- 변수 : 상수로 정의한다.
- 클래스 : 상속을 허용하지않는다
- 메소드 : 오버라이딩을 허용하지 않는다.

static : 메모리공간에 단 하나만 존재하는 어디서나 접근 가능한 변수, 인스턴스 간 메모리 공유가 필요할때 정의한다.
- 클래스 내부 : 언제든지 직접접근 가능
- 클래스 외부 : 인스턴스 이름이나 클래스 이름을 통해 접근가능
(인스턴스 생성 이전에 클래스 이름으로 static 변수에 접근이 가능한 이유는 클래스가 메모리 공간에 올라가는 순간 초기화되기 때문)

static final - 참조의 용도로만 선언되는 변수를 static final로 선언.

refer : https://goodncuteman.tistory.com/4

kyeah01 replied:
476409925fb4830e286314bce9d4e704dea9c174 🚀

kyeah01 replied:
확인하시고 괜찮다 싶으심 resolve 부탁쓰ㅎㅎㅎ

kimevanjunseok commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,46 @@
+public class Main {
+    public static void main (String[] args) {
+        Scanner sc = new Scanner(System.in);
+        System.out.println("경주할 자동차 이름을 입력하세요(이름은 쉼표(,)를 기준으로 구분).");
```
MVC 패턴이라고 하징
저런 프린트는 View에 속해용. 그래서 이정도 미션할 때
M = domain, V = view, C = Application 이렇게 나눠용

kimevanjunseok replied:
MVC도 디자인 패턴임당...

kimevanjunseok replied:
이 미션에서 MVC 적용해보는 것도 나쁘지않아!!

kyeah01 replied:
M : model, 무엇을 할지 결정한다.
- 사용자가 편집하기 원하는 모든 정보를 가지고있어야 한다.
- 뷰나 컨트롤러의 어떤 정보도 알지못해야 한다.
- 이벤트(변경)이 일어나면 이벤트 통지해야 함.
- 재사용가능해야 하며 인터페이스의 차이에 따라 변하지 않아야 함.

V : view, 무엇을 보여줄 것인지 결정한다.
- model 정보를 caching하면 안됨(받아서 처리하고 끝)
- 다른 구성요소를 몰라야 함. 받아서 정보활용, 끝.
- 이벤트(변경)이 일어나면 이벤트 통지해야 함.
- 재사용이 가능해야 함.

C : controller, 어떻게 할 것인지 결정한다.
- 모델과 뷰를 알아야 함. 둘 사이에 중재해야 함.
- 모델이나 뷰의 변경을 모니터링 할 것.

MVC를 활용하면 
pros - 확장성, 유연성 증가. 중복성 감소.
cons - 뷰와 모델이 의존성을 가짐. 기능 추가시 비대해질 우려가 있음.

kyeah01 replied:
refer : 
https://medium.com/@jang.wangsu/1d74fac6e256
https://m.blog.naver.com/jhc9639/220967034588

kimevanjunseok commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,46 @@
+    public static void main (String[] args) {
+        Scanner sc = new Scanner(System.in);
+        System.out.println("경주할 자동차 이름을 입력하세요(이름은 쉼표(,)를 기준으로 구분).");
+        String[] carNames = sc.nextLine().split(",");
```
보통 java api를 사용해 예를들면 List<String> 이런것
이유는 저런 String[]은 오류가 발생하면 런타임이지만 java api를 사용하면 컴파일 오류가 발생!!
그래서 String[]같은 것을 사용하면 서버는 돌아가서 문제인식을 못하지만 java api는 컴파일자체가 안되서 서버가 안 켜져

kyeah01 replied:
java.util.List 말하는거 맞남??

kyeah01 replied:
검색좀 해보니까 다들
```
List<String> list = new ArrayList<String>();
```
이런식으로만 활용하는데 왜
```
List <String> list = new List<String>();
```
으로는 활용 안하는지 물어봐도 될까리????ㅠ

kimevanjunseok commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,46 @@
+        System.out.println("시도할 회수는 몇회인가요?");
+        int count = sc.nextInt();
+
+        Car[] cars = new Car[countofCar];
```
그럼 java api를 사용하면 이건 List\<Car\>가 되는 거지

kimevanjunseok commented on src/main/java/domain/Car.java:
```diff
@@ -1,12 +1,10 @@
 public class Car {
-    private final String name;
-    private int position = 0;
+    public final String name;
```
캡슐화를 공부하면 왜 private를 쓰는지 알 수 있엉!!

kimevanjunseok replied:
https://mainpower4309.tistory.com/7

kimevanjunseok reviewed (commented):
간단한 것만 말했엉!!! 계속해서 말해줄겡!!!

kyeah01 committed:
200615 | Refactor : devide method && set variable
- devide 'get car name', 'get race time' method
- set 'scanner' static final variable

kyeah01 committed:
M-V-C model apply
- MVC apply : domain, view, controller
- create Cars : first-calss obj
- devide inputview / outputview
